jvm/sargon-android/src/test/java/com/radixdlt/sargon/TransactionManifestTest.kt

```diff
@@ -7125,227 +294,97 @@ class TransactionManifestTest: SampleTestable<TransactionManifest> {
-                Decimal("1337")
-            ;
-            TAKE_FROM_WORKTOP
-                Address("resource_rdx1tknxxxxxxxxxradxrdxxxxxxxxx009923554798xxxxxxxxxradxrd")
-                Decimal("1337")
-                Bucket("bucket1")
-            ;
-            CALL_METHOD
-                Address("account_rdx16yf8jxxpdtcf4afpj5ddeuazp2evep7quuhgtq28vjznee08master")
-                "try_deposit_or_abort"
-                Bucket("bucket1")
-                Enum<0u8>()
-            ;
+        val instructions = manifest("create_pool")
 
-        """.trimIndent()
-        val unmodifiedManifest = TransactionManifest.init(
-            instructionsString = unmodified,
-            networkId = NetworkId.MAINNET
-        )
+        assertFalse(instructions.instructionsString.contains("lock_fee"))
 
-        val modifiedManifest = unmodifiedManifest.modifyLockFee(
-            addressOfFeePayer = AccountAddress.init(
-                validatingAddress = "account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease"
-            ),
-            fee = 0.61f.toDecimal192(),
+        val modified = instructions.modifyLockFee(
+            addressOfFeePayer = AccountAddress.sampleStokenet(),
+            fee = 195.toDecimal192()
         )
-        val expectedModified = """
-            CALL_METHOD
-                Address("account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease")
-                "lock_fee"
-                Decimal("0.61")
-            ;
-            CALL_METHOD
-                Address("account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease")
-                "withdraw"
-                Address("resource_rdx1tknxxxxxxxxxradxrdxxxxxxxxx009923554798xxxxxxxxxradxrd")
-                Decimal("1337")
-            ;
-            TAKE_FROM_WORKTOP
-                Address("resource_rdx1tknxxxxxxxxxradxrdxxxxxxxxx009923554798xxxxxxxxxradxrd")
-                Decimal("1337")
-                Bucket("bucket1")
-            ;
-            CALL_METHOD
-                Address("account_rdx16yf8jxxpdtcf4afpj5ddeuazp2evep7quuhgtq28vjznee08master")
-                "try_deposit_or_abort"
-                Bucket("bucket1")
-                Enum<0u8>()
-            ;
 
-        """.trimIndent()
+        with(modified.instructionsString) {
+            assertTrue(contains("lock_fee"))
+            assertTrue(contains("195"))
+            assertTrue(contains(AccountAddress.sampleStokenet().string))
+        }
+    }
 
-        assertEquals(expectedModified, modifiedManifest.instructionsString)
+    private fun String.occurrences(substring: String): Int {
+        val matcher = Pattern.compile(substring).matcher(this)
+        var counter = 0
+        while (matcher.find()) {
+            counter++
+        }
+        return counter
     }
+
+    private fun manifest(name: String) = TransactionManifest.init(
+        instructionsString = File(
```

Comment:
I had to do this in Swift since accessing files in Swift Tests has really bad support... I had hoped Kotlin offers better support, but maybe not? 😫

Reply:
It has better support but you need to put the files into the `resources` folder. Since these files exist elsewhere it was the only option.

Reply:
Maybe I can but again, it will be a file being opened and read. Nothing more fancy I think.